packages/DoctrineCodeQuality/src/Rector/Class_/InitializeDefaultEntityCollectionRector.php

```diff
@@ -0,0 +1,190 @@
+use PhpParser\Node\Stmt\Expression;
+use Rector\BetterPhpDocParser\Contract\Doctrine\ToManyTagNodeInterface;
+use Rector\BetterPhpDocParser\PhpDocNode\Doctrine\Class_\EntityTagValueNode;
+use Rector\Doctrine\ValueObject\DoctrineClass;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+/**
+ * @see https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/best-practices.html#initialize-collections-in-the-constructor
+ *
+ * @see \Rector\DoctrineCodeQuality\Tests\Rector\Class_\InitializeDefaultEntityCollectionRector\InitializeDefaultEntityCollectionRectorTest
+ */
+final class InitializeDefaultEntityCollectionRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Initialize collection property in Entity constructor', [
+            new CodeSample(
+                <<<'PHP'
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ */
+class SomeClass
+{
+    /**
+     * @ORM\OneToMany(targetEntity="MarketingEvent")
+     */
+    private $marketingEvents = [];
+}
+PHP
+                ,
+                <<<'PHP'
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ */
+class SomeClass
+{
+    /**
+     * @ORM\OneToMany(targetEntity="MarketingEvent")
+     */
+    private $marketingEvents = [];
+
+    public function __construct()
+    {
+        $this->marketingEvents = new ArrayCollection();
+    }
+}
+PHP
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $classPhpDocInfo = $this->getPhpDocInfo($node);
+        if ($classPhpDocInfo === null) {
+            return null;
+        }
+
+        if (! $classPhpDocInfo->getByType(EntityTagValueNode::class)) {
+            return null;
+        }
+
+        $toManyPropertyNames = $this->resolveToManyPropertyNames($node);
+        if ($toManyPropertyNames === []) {
+            return null;
+        }
+
+        $constructClassMethod = $node->getMethod('__construct');
+        $assigns = $this->createAssignsOfArrayCollectionsForPropertyNames($toManyPropertyNames);
+
+        if ($constructClassMethod === null) {
+            $constructClassMethod = $this->nodeFactory->createPublicMethod('__construct');
+            $constructClassMethod->stmts = $assigns;
+
+            $node->stmts = array_merge((array) $node->stmts, [$constructClassMethod]);
+        } else {
+            $assigns = $this->filterOutExistingAssigns($constructClassMethod, $assigns);
+
+            // all properties are initialized → skip
+            if ($assigns === []) {
+                return null;
+            }
+
+            $constructClassMethod->stmts = array_merge($assigns, (array) $constructClassMethod->stmts);
+        }
```

Comment:
This whole section should be decoupled to own service.

- add these nodes to method with name x, if they're not already there